### PR TITLE
Added docstrings and type hints to oscillator.py

### PR DESCRIPTION
**Pull request by kaleb2**

### CODE REVIEW

File: src/oscillator.py [hunk 1]
```diff
@@ -101,26 +145,32 @@ def generate_wave(self):
...
+    def generate_wave(self) -> np.ndarray:
+        """Generates a triangle wave"""
+
+        samples: np.ndarray = np.empty(
+            int(self._sample_rate * self._duration), dtype=float
+        )
 
         # half-period
-        hp = (1 / self._frequency) / 2
+        hp: float = (1 / self._frequency) / 2
```
Comment (per-sephone) [inline src/oscillator.py]: we could change the variable name "hp" to half_period here and remove the comment

File: src/oscillator.py [hunk 2]
```diff
@@ -101,26 +145,32 @@ def generate_wave(self):
...
 
         # double the amplitude
-        da = self._amplitude * 2
+        da: float = self._amplitude * 2
```
Comment (per-sephone) [inline src/oscillator.py]: we could change the variable name "da" here to "double_amplitude"

**Review by per-sephone — APPROVED** [review]
I added a couple comments about updating variable names. Everything else looks great!

**Review by jasonandmonte — APPROVED** [review]
Looks good to me!

**Pull request merged**